golang/pkg/events/events_test.go

```diff
@@ -28,22 +30,36 @@ func TestToEvent(t *testing.T) {
 		},
 	}
 
+	id := "id"
+	source := "wandalorian"
+	time := time.Now()
+
 	data, err := anypb.New(&event)
 
+	attr := events.CloudEventAttributeValue{
+		Attr: &events.CloudEventAttributeValue_CeTimestamp{
+			CeTimestamp: timestamppb.New(time),
+		},
+	}
+
 	assert.NoError(t, err)
 	ce := events.CloudEvent{
-		Id:          "id",
-		Source:      "wandalorian",
+		Id:          id,
+		Source:      source,
 		SpecVersion: "1.0",
 		Type:        string(event.ProtoReflect().Descriptor().FullName()),
 		Data: &events.CloudEvent_ProtoData{
 			ProtoData: data,
 		},
+		Attributes: map[string]*events.CloudEventAttributeValue{
+			"time": &attr,
+		},
 	}
 
 	rawCe, err := proto.Marshal(&ce)
 	assert.NoError(t, err)
-	encodedEvent, err := events.ToEvent(&event, "wandalorian", "id")
+
+	encodedEvent, err := events.ToEvent(&event, events.WithId(&id), events.WithSource(&source), events.WithTime(&time))
```

Comment:
It would be nice to have a test without the options, and simply checks that it doesn't raise any error.